fcrypt: failing test case if current directory is not writable

# Describe what you wanted to do

Run `kdb run_all` with installed Elektra on my machine.

Seems to *not* occur on http://build.libelektra.org:8080/job/elektra-local-installation/
neither on a fresh installation:

- Install jessie (e.g. `deboostrap jessie jessie`).
- add `deb     [trusted=yes] http://debian-stable.libelektra.org/elektra-stable/ jessie main` to /etc/apt/sources.list
- `apt-get update`
- `apt-get install libelektra4-bin libelektra-test`
- `kdb run_all`

But it is always reproducable on my machine, even when wiping out all keys of kdb or when running tests with `env -i`. Same behavior with `KDB=kdb-static kdb-static run_all` and `KDB=kdb-full kdb-full run_all`.

@petermax2 Can you reproduce it?

# Describe what you expected

That all test cases run successfully.

# Describe what actually happened

`error: testmod_fcrypt fails` if you run it with `kdb run_all` (I tried it several times).

It does not fail when you run it separately (`kdb testmod_fcrypt`, I tried it several hundret times).

# System Information

- Elektra Version: master
- gpg (GnuPG) 2.0.26
- libgcrypt 1.6.3

# Further Log Files and Output
```
--- running testmod_fcrypt ---


FCRYPT       TESTS
==================

/home/jenkins/workspace/workspace/elektra-git-buildpackage-jessie/libelektra/src/plugins/fcrypt/testmod_fcrypt.c:50: error in writeTestFile: can not write to temporary file
/home/jenkins/workspace/workspace/elektra-git-buildpackage-jessie/libelektra/src/plugins/fcrypt/testmod_fcrypt.c:156: error in test_file_operations: kdb set failed
/home/jenkins/workspace/workspace/elektra-git-buildpackage-jessie/libelektra/src/plugins/fcrypt/testmod_fcrypt.c:68: error in isTestFileCorrect: can not read from temporary file
/home/jenkins/workspace/workspace/elektra-git-buildpackage-jessie/libelektra/src/plugins/fcrypt/testmod_fcrypt.c:160: error in test_file_operations: kdb get (pregetstorage) failed
/home/jenkins/workspace/workspace/elektra-git-buildpackage-jessie/libelektra/src/plugins/fcrypt/testmod_fcrypt.c:68: error in isTestFileCorrect: can not read from temporary file
/home/jenkins/workspace/workspace/elektra-git-buildpackage-jessie/libelektra/src/plugins/fcrypt/testmod_fcrypt.c:161: error in test_file_operations: file content could not be restored during decryption
/home/jenkins/workspace/workspace/elektra-git-buildpackage-jessie/libelektra/src/plugins/fcrypt/testmod_fcrypt.c:164: error in test_file_operations: kdb get (postgetstorage) failed
/home/jenkins/workspace/workspace/elektra-git-buildpackage-jessie/libelektra/src/plugins/fcrypt/testmod_fcrypt.c:68: error in isTestFileCorrect: can not read from temporary file

fcrypt RESULTS: 23 test(s) done. 8 error(s).
error: testmod_fcrypt

....


+ exit 0
Following test cases failed:
testmod_fcrypt
run_all RESULTS: 667 test(s) done 1 error(s).
```

--- a/src/plugins/fcrypt/testmod_fcrypt.c
+++ b/src/plugins/fcrypt/testmod_fcrypt.c
@@ -30,10 +30,16 @@ static const kdb_octet_t testContent[] = { 0x01, 0x02, 0xCA, 0xFE, 0xBA, 0xBE, 0
  */
 static char * getTemporaryFileName ()
 {
-	const size_t newFileAllocated = strlen (TEST_FILE) + 7;
+	const char * tmpvar = getenv ("TMPDIR");
+	if (!tmpvar)
+	{
+		tmpvar = "/tmp";
+	}
+
+	const size_t newFileAllocated = strlen (tmpvar) + 1 + strlen (TEST_FILE) + 7;
 	char * newFile = elektraMalloc (newFileAllocated);
 	if (!newFile) return NULL;
-	snprintf (newFile, newFileAllocated, "%sXXXXXX", TEST_FILE);
+	snprintf (newFile, newFileAllocated, "%s/%sXXXXXX", tmpvar, TEST_FILE);
 	mkstemp (newFile);
 	return newFile;
 }

